### PR TITLE
feat!: compact block propagation

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---

- Changes the propagated new block message to contain `<block_header, coinbase, excess_sigs>`
- Reconciles incoming new block with known mempool transactions by excess signature
- Requests, if required, missing transactions from peer
- Failing that, request the full block from the peer
- Reduce (deallocate) mempool memory when transactions are removed 
- Replace real time-based reorg pool expiry with block height-based expiry
- Clean up deprecated mempool messaging handlers etc
- Ensure DHT connectivity keeps hold of connection handles 

Motivation and Context
---
In the majority of cases, the inputs, outputs and kernels to be inserted in a block are already known and exist in the mempool. Previously, a propagated new block message would just contain the block hash, requiring the node request the full block from the peer. This implementation provides the minimum block data required to assemble a valid block and propagates that message (typically less than 100kb).

How Has This Been Tested?
---
Updated tests, existing block propagation tests
Manually: 3 base nodes, 1 wallet and 4 blocks worth of transactions in the mempool on dibbler

### CODE REVIEW

File: base_layer/core/src/base_node/comms_interface/comms_request.rs [hunk 1]
```diff
@@ -83,16 +86,16 @@ impl Display for NodeCommsRequest {
...
             FetchMatchingUtxos(v) => write!(f, "FetchMatchingUtxos (n={})", v.len()),
             FetchMatchingTxos(v) => write!(f, "FetchMatchingTxos (n={})", v.len()),
             FetchMatchingBlocks(range) => {
                 write!(f, "FetchMatchingBlocks ({:?})", range)
             },
             FetchBlocksByHash(v) => write!(f, "FetchBlocksByHash (n={})", v.len()),
-            FetchBlocksWithKernels(v) => write!(f, "FetchBlocksWithKernels (n={})", v.len()),
-            FetchBlocksWithUtxos(v) => write!(f, "FetchBlocksWithUtxos (n={})", v.len()),
+            FetchBlocksByKernelExcessSigs(v) => write!(f, "FetchBlocksWithKernels (n={})", v.len()),
+            FetchBlocksByUtxos(v) => write!(f, "FetchBlocksWithUtxos (n={})", v.len()),
```
Comment (delta1) [inline base_layer/core/src/base_node/comms_interface/comms_request.rs]: very minor but these strings don't match the variants any more

File: base_layer/core/src/mempool/reorg_pool/reorg_pool.rs [hunk 2]
```diff
@@ -20,102 +20,326 @@
...
+    config: ReorgPoolConfig,
+    key_counter: usize,
+    tx_by_key: HashMap<TransactionId, Arc<Transaction>>,
+    txs_by_signature: HashMap<PrivateKey, Vec<TransactionId>>,
+    txs_by_height: HashMap<u64, Vec<TransactionId>>,
 }
 
 impl ReorgPool {
-    /// Create a new ReorgPool with the specified configuration
+    /// Create a new ReorgPoolwith the specified configuration
```
Comment (delta1) [inline base_layer/core/src/mempool/reorg_pool/reorg_pool.rs]:
```suggestion
    /// Create a new ReorgPool with the specified configuration
```

File: base_layer/core/src/mempool/reorg_pool/reorg_pool.rs [hunk 3]
```diff
@@ -20,102 +20,326 @@
...
-        }
+        Self { expiry_height: 5 }
     }
 }
 
+type TransactionId = usize;
+
 /// The ReorgPool consists of all transactions that have recently been added to blocks.
 /// When a potential blockchain reorganization occurs the transactions can be recovered from the ReorgPool and can be
 /// added back into the UnconfirmedPool. Transactions in the ReOrg pool have a limited Time-to-live and will be removed
```
Comment (delta1) [inline base_layer/core/src/mempool/reorg_pool/reorg_pool.rs]: minor: doc comment needs an update as it still references TTL

File: base_layer/core/src/base_node/comms_interface/outbound_interface.rs [hunk 4]
```diff
@@ -70,6 +75,27 @@ impl OutboundNodeCommsInterface {
         }
     }
 
+    /// Fetch the Blocks corresponding to the provided block hashes from a specific base node. The requested blocks
+    /// could be chain blocks or orphan blocks.
```
Comment (delta1) [inline base_layer/core/src/base_node/comms_interface/outbound_interface.rs]: minor: this doc comment is duplicate/unrelated

**Review by delta1 — COMMENTED** [review]
Awesome! 👏 

Some very minor comments while I build and test

File: base_layer/core/src/base_node/comms_interface/outbound_interface.rs [hunk 5]
```diff
@@ -75,7 +75,7 @@ impl OutboundNodeCommsInterface {
         }
     }
 
-    /// Fetch the Blocks corresponding to the provided block hashes from a specific base node. The requested blocks
+    /// Fetch the Blocks corresponding to the provided excess_sig from the given peer `NodeId`. The requested blocks
```
Comment (delta1) [inline base_layer/core/src/base_node/comms_interface/outbound_interface.rs]:
```suggestion
    /// Fetch the transactions corresponding to the provided excess_sigs from the given peer `NodeId`. 
```

File: base_layer/core/src/base_node/comms_interface/outbound_interface.rs [hunk 6]
```diff
@@ -75,7 +75,7 @@ impl OutboundNodeCommsInterface {
...
     /// could be chain blocks or orphan blocks.
```
Comment (delta1) [inline base_layer/core/src/base_node/comms_interface/outbound_interface.rs]:
```suggestion
```

**Review by stringhandler — APPROVED** [review]
So keen to try this out

**Pull request merged**